core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/generateCode.kt

```diff
@@ -1,14 +1,89 @@
 package org.jetbrains.kotlinx.dataframe.api
 
 import org.jetbrains.kotlinx.dataframe.DataFrame
+import org.jetbrains.kotlinx.dataframe.annotations.ColumnName
+import org.jetbrains.kotlinx.dataframe.annotations.DataSchema
 import org.jetbrains.kotlinx.dataframe.codeGen.CodeGenerator
 import org.jetbrains.kotlinx.dataframe.codeGen.MarkerVisibility
 import org.jetbrains.kotlinx.dataframe.codeGen.NameNormalizer
+import org.jetbrains.kotlinx.dataframe.columns.ColumnGroup
+import org.jetbrains.kotlinx.dataframe.columns.FrameColumn
+import org.jetbrains.kotlinx.dataframe.documentation.DocumentationUrls
+import org.jetbrains.kotlinx.dataframe.documentation.ExcludeFromSources
 import org.jetbrains.kotlinx.dataframe.impl.codeGen.from
 import org.jetbrains.kotlinx.dataframe.schema.DataFrameSchema
 
+// region Docs
+
+/**
+ * Generates a [CodeString] containing generated [@DataSchema][DataSchema] $[TYPES]
+ * for the given $[RECEIVER]
+ * (including all nested [frame columns][FrameColumn] and [column groups][ColumnGroup]).
+ *
+ * These generated declarations can also be called "markers".
+ *
+ * This is useful when working with the compiler plugin in cases where the schema
+ * cannot be inferred automatically from the source.
+ *
+ * This function is a simplified wrapper for the more advanced and customizable
+ * [CodeGenerator] API.
+ * For more customizability, have a look at [CodeGenerator.create()][CodeGenerator.create].
+ *
+ * For more information: {@include [DocumentationUrls.DataSchemaGeneration]}
+ *
+ * @return [CodeString] – A value class wrapper for [String], containing
+ *   the generated Kotlin code of data schema declarations (markers).
+ */
+@ExcludeFromSources
+private interface CommonGenerateCodeDocs {
+
+    // "interfaces" or "data classes"
+    interface TYPES
+
+    // "DataFrameSchema" or "DataFrame"
+    interface RECEIVER
+}
+
+@ExcludeFromSources
+private interface Params {
+
+    /** @param markerName The base name to use for generated data schema declarations (markers). */
+    interface MarkerName
+
+    /** @include [MarkerName] If not specified, generates a name from type [T]. */
+    interface MarkerNameOptional
+
+    /** @param fields Whether to generate fields (`val ...:`) inside the generated data schema declarations (markers). */
+    interface Fields
+
+    /** @param extensionProperties Whether to generate extension properties in addition to data schema declarations (markers). */
```

Comment:
Here we should say that if either compiler plugin or ksp are used in the project, extension properties will conflict with generated ones. 
Soo, same as with fields, does it even make sense to have this option..

Reply:
Let's keep it but switch default to false, i didn't know about it :) But for now we'll recommend using generateCode with plugin and properties will surely conflict